ikenna-portfolio/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,35 +14,89 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.QueryResultList;
 import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import com.google.sps.data.Metadata;
+import com.google.sps.data.Metadata.Search;
+import com.google.sps.database.CommentDatabase;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+import java.lang.*;
+import java.util.*;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns JSON content. */
+/** Servlet that creates comments from form data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> msgs;
+
+  private CommentDatabase database;
+  private List<Comment> comments;
```

Comment:
I believe comments and metadata fields caused the issue with sorting, which is why it's better to keep servlets stateless. The two data requests when doing sorting is causing thread safety issues.
A thread from stackoverflow talking about servlet thread safety: https://stackoverflow.com/questions/3106452/how-do-servlets-work-instantiation-sessions-shared-variables-and-multithreadi

Reply:
done